### DISCUSSION TITLE
How can I get Neodev to also work on project local ".nvimrc.lua" files?

### DISCUSSION THREAD

**Discussion by Hubro** [folke/lua-dev.nvim#160] [Q&A]
I have this line in my `init.lua`:

```lua
if vim.fn.filereadable(".nvimrc.lua") == 1 then
  vim.cmd("source .nvimrc.lua")
end
```

This lets me set project specific key maps, autocommands etc.

However, Neodev doesn't set up the Lua LSP for these files, so I don't get auto completions, which makes these files annoying to edit.

How would I go about getting Neodev to start up when I open any `.nvimrc.lua` file?

Alternatively, I don't really use Lua for anything other than Neovim config, could I configure Neodev to run for all Lua files?

**Comment by paul-hansen** [comment]
You can make it enabled for any project by using the `override` option and enable it for every directory.
```lua
use {
    "folke/neodev.nvim",
    config = function()
        require("neodev").setup {
            override = function(root_dir, library)
                -- enable for any project so we can use it in .nvim.lua files
                library.enabled = true
                library.plugins = true
            end,
        }
    end
}
```
To save some work on non-lua projects, you could probably have it check for lua files in the root_dir and only enable it if there are.
